src/domain/position/timeout-reserved.js

```diff
@@ -0,0 +1,169 @@
+          _handleParticipantPositionChange(runningPosition, transferAmount, transferId, accumulatedPositionReservedValue)
+        Logger.isDebugEnabled && Logger.debug(`processPositionTimeoutReservedBin::participantPositionChange: ${JSON.stringify(participantPositionChange)}`)
+        runningPosition = updatedRunningPosition
+        binItem.result = { success: true }
+        participantPositionChanges.push(participantPositionChange)
+        transferStateChanges.push(transferStateChange)
+        accumulatedTransferStatesCopy[transferId] = transferStateId
+        resultMessages.push({ binItem, message: resultMessage })
+      }
+    }
+  }
+
+  return {
+    accumulatedPositionValue: runningPosition.toNumber(),
+    accumulatedTransferStates: accumulatedTransferStatesCopy, // finalized transfer state after fulfil processing
+    accumulatedFxTransferStates: accumulatedFxTransferStatesCopy, // finalized transfer state after fx fulfil processing
+    accumulatedPositionReservedValue, // not used but kept for consistency
+    accumulatedTransferStateChanges: transferStateChanges, // transfer state changes to be persisted in order
+    accumulatedFxTransferStateChanges: fxTransferStateChanges, // fx-transfer state changes to be persisted in order
+    accumulatedPositionChanges: participantPositionChanges, // participant position changes to be persisted in order
+    notifyMessages: resultMessages, // array of objects containing bin item and result message. {binItem, message}
+    followupMessages // array of objects containing bin item, message key and followup message. {binItem, messageKey, message}
+  }
+}
+
+const _constructTimeoutReservedResultMessage = (binItem, transferId, payeeFsp, payerFsp) => {
+  const headers = { ...binItem.message.value.content.headers }
+  headers[Enum.Http.Headers.FSPIOP.DESTINATION] = payeeFsp
+  headers[Enum.Http.Headers.FSPIOP.SOURCE] = payerFsp
+  delete headers['content-length']
+
+  // Create a FSPIOPError object for timeout payee notification
+  const fspiopError = ErrorHandler.Factory.createFSPIOPError(
+    ErrorHandler.Enums.FSPIOPErrorCodes.TRANSFER_EXPIRED,
+    null,
+    null,
+    null,
+    null
+  ).toApiErrorObject(Config.ERROR_HANDLING)
+
+  const state = Utility.StreamingProtocol.createEventState(
+    Enum.Events.EventStatus.FAILURE.status,
+    fspiopError.errorInformation.errorCode,
+    fspiopError.errorInformation.errorDescription
+  )
+
+  // Create metadata for the message, associating the payee notification
+  // with the position event timeout-reserved action
+  const metadata = Utility.StreamingProtocol.createMetadataWithCorrelatedEvent(
+    transferId,
+    Enum.Kafka.Topics.POSITION,
+    Enum.Events.Event.Action.TIMEOUT_RESERVED,
+    state
+  )
+  const resultMessage = Utility.StreamingProtocol.createMessage(
+    transferId,
+    payeeFsp,
+    payerFsp,
+    metadata,
+    headers, // Unsure what headers need to be passed here
```

Comment:
:thinking: couldn't tell from the non batch code what the headers for this are.

Reply:
The ml-api-adapter deals with notifying the payee and payer, so headers don't matter and the adapter will construct the proper headers!
https://github.com/mojaloop/ml-api-adapter/blob/main/src/handlers/notification/index.js#L545

As long as the to and from are the payer and payee, which they have been and the action is timeout-reserved ml adapter will notify both.